server/tools/peertube-import-videos.ts

```diff
@@ -68,30 +106,37 @@ async function run (url: string, user: UserInfo) {
   const options = [ '-j', '--flat-playlist', '--playlist-reverse' ]
   youtubeDL.getInfo(program[ 'targetUrl' ], options, processOptions, async (err, info) => {
     if (err) {
-      console.log(err.message)
+      log.error(err.message)
       process.exit(1)
     }
 
     let infoArray: any[]
 
     // Normalize utf8 fields
     if (Array.isArray(info) === true) {
-      infoArray = info.map(i => normalizeObject(i))
+      if (program[ 'first' ]) {
+        infoArray = info.slice(0, program[ 'first' ]).map(i => normalizeObject(i))
+      } else if (program[ 'last' ]) {
+        infoArray = info.slice(0 - program[ 'last' ]).map(i => normalizeObject(i))
+      } else {
+        infoArray = info.map(i => normalizeObject(i))
+      }
```

Comment:
I may nitpick here: I guess we shouldn't repeat the `.map(i => normalizeObject(i))`.

```js
if (Array.isArray(info) === true) {
  if (program[ 'first' ]) {
    infoArray = info.slice(0, program[ 'first' ])
  } else if (…) {
    infoArray = info.slice(- program[ 'last' ])
  } else {
    infoArray = info
  }
  infoArray = info.map(i => normalizeObject(i))
}
```

Also we can use this trick to avoid the `if (Array.isArray(info) === true) {` condition (`concat` can be used to force a value to be an array):
```js
infoArray = [].concat(info);
if (program[ 'first' ]) {
  infoArray = infoArray.slice(0, program[ 'first' ])
} else if (program[ 'last' ]) {
  infoArray = infoArray.slice(- program[ 'last' ])
}
infoArray = infoArray.map(i => normalizeObject(i))
```

Reply:
The former, sure, that is cleaner. The latter, well, it works but is isn't cleaner or clearer than the original. I don't really care though, as long as it works and looks somewhat maintainable.